video_store.py

```diff
@@ -0,0 +1,146 @@
+import requests
+from datetime import datetime
+
+class VideoStore:
+    def __init__(self, url="https://retro-video-store-api.herokuapp.com", selected_video=None, selected_customer=None):
+        self.url = url
+        self.selected_video = selected_video
+        self.selected_customer = selected_customer
+        
+
+##### VIDEO #####
+    def create_video(self, title="Default Title", release_date="Default Date", total_inventory=0):
+        query_params = {
+            "title": title,
+            "release_date": release_date,
+            "total_inventory": total_inventory
+        }
+        response = requests.post(self.url+"/videos",json=query_params)
+        return response.json()
+
+    def update_video(self, title=None, release_date="Default Date", total_inventory=0):
+        if not title:
+            title = self.selected_video["title"]
+        if not release_date:
+            release_date = self.selected_video["release_date"]
+        if not total_inventory:
+            total_inventory = self.selected_video["total_inventory"]
+
+        query_params = {
+        "title": title,
+        "release_date": release_date,
+        "total_inventory": total_inventory}
+
+        response = requests.put(self.url+f"/videos/{self.selected_video['id']}", 
+                                json=query_params)
+        print("response:", response)
+        self.selected_video = response.json()["video"]
+        return response.json()    
+
+    def delete_video(self):
+        response = requests.delete(self.url+f"/videos/{self.selected_video['id']}")
+        self.selected_video = None
+        return response.json()
+
+    def get_videos_list(self):
+        response = requests.get(self.url+"/videos")
+        return response.json()
+
+    def get_single_video(self, title=None, id=None):
+        for video in self.get_videos_list():
+            if title:
+                if video["title"]==title:
+                    id = video["id"]
+                    self.selected_video = video
+            elif id == video["id"]:
+                self.selected_video = video
+
+        if self.selected_video == None:
```

Comment:
There is a bug in the logic here.  Consider this sequence:
1. This function is called and a video is found.  selected_video is set to that video, call it VideoA
2. The function is called and no video is found.  This test will fail because selected_video is set to VideoA
3. The request will attempt to get the video with either a non-existent ID or None, which will return a 404 response.

To fix this, self.selected_video must be reset at the beginning of this function.